geniza/corpus/views.py

```diff
@@ -29,51 +23,78 @@ def get_queryset(self, *args, **kwargs):
 # --------------- Publish CSV to sync with old PGP site --------------------- #
 
 
-def parse_edition_string(editions):
+def old_pgp_edition(editions):
+    """output footnote and source information in a format similar to
+    old pgp metadata editor/editions."""
     if editions:
+        # label as translation if edition also supplies translation;
+        # include url if any
         edition_list = [
-            "trans. " + fn.display().strip(".")
-            if Footnote.TRANSLATION in fn.doc_relation
-            else "ed. " + fn.display().strip(".")
+            "%s%s%s"
+            % (
+                "and trans. " if Footnote.TRANSLATION in fn.doc_relation else "",
+                fn.display().strip("."),
+                " %s" % fn.url if fn.url else "",
+            )
             for fn in editions
         ]
+        # combine multiple editons as Ed. ...; also ed. ...
+        return "".join(["Ed. ", "; also ed. ".join(edition_list), "."])
 
-        edition_list = [
-            s + f" {fn.url}" if fn.url else s for s, fn in zip(edition_list, editions)
-        ]
-
-        return_str = "; also ".join(edition_list)
-
-        # TODO: omg there *has* to be a better way to just capitalize the first
-        #  letter of a string in python.
-        return return_str[0].upper() + return_str[1:] + "."
     return ""
 
 
-def tabulate_queryset(queryset):
+def old_pgp_tabulate_data(queryset):
+    """Takes a :class:`~geniza.corpus.models.Document` queryset and
+    yields rows of data for serialization as csv in :method:`pgp_metadata_for_old_site`"""
     # NOTE: This logic assumes that documents will always have a fragment
     for doc in queryset:
         primary_fragment = doc.textblock_set.first().fragment
+        num_fragments = len([tb for tb in doc.textblock_set.all() if tb.certain])
```

Comment:
the naming of `num_fragments` suggests this counts fragments, but doesn't it count `TextBlock`s? currently we can have multiple `TextBlock`s between a single `Document` and `Fragment` pair.

Reply:
I don't actually care how many they are, I just wanted a quick check to see if there is more than one — it's used to determine whether we put the combined shelfmark in the join column; I thought it would be faster to count vs  generating the full shelfmark and checking for the presence of "+"

Reply:
right, but in the case described above, wouldn't that be erroneously counted as a join? we might not actually have any at the moment, not sure

Reply:
oh! I see what you mean. yes, it might ... I don't think we have any like that currently. Suggestion for a quick revision that would be more future proof?

Reply:
i think it's just the same logic that gets used to generate the shelfmark – the only surefire way to know if something is a join is if it has more than one unique, certain fragment. order doesn't matter here though, so you could do:
```py
set([tb.fragment for tb in doc.textblock_set.all() if tb.certain])
```